### ISSUE TITLE
Can translate swf file to any video format?

### ISSUE THREAD

**Issue by gzjackyguan** [lightspark/lightspark#434]
I want to use this tools to translate swf file to any format video。perhaps in linux and have a sdk api。Thanks！

**Comment by wyatt8740** [comment]
I think I might be able to expound upon what is being asked for, perhaps.

Gnash had/has a `dump-gnash` tool, which could be used to do a raw video dump of a flash file that could be processed by ffmpeg. Even if a flash file could not render and record in real-time at a given resolution, gnash would methodically go through frame by frame and record progress so that the resulting video was at the smooth original frame rate.

`dump-gnash` supported three CLI arguments in addition to the normal gnash ones:

```
  -D arg                        Video dump file (raw format) and optional video FPS (@<num>)
  -S arg                        Number of milliseconds to sleep between advances
  -T arg                        Trigger expression to start dumping
```
Also `-1` would make it close after one loop, which was handy. Not sure what constituted a trigger expression; I always just had it start when the flash did.

An example CLI taken from stack exchange (I always would look there when I wanted to record something for reference):
```
dump-gnash -1 \
           -D /tmp/out.raw@30 \
           -A /tmp/out.wav \
           ...other args...
           http://example.com/some.swf
```

Or how I would usually do it (from memory, at least!):
```
mkfifo video.raw
# pipe video.raw to ffmpeg; don't worry about audio at the moment because I have room for uncompressed PCM and it requires less effort if I just re-combine them later
dump-gnash -1 \
           -D video.bgra@framerate \
           -A audio.wav
           http://example.com/some.swf
```

…in another shell…
```
ffmpeg -f rawvideo -vcodec rawvideo -s WIDTHxHEIGHT -r framerate -pix_fmt bgra -i video.bgra -vcodec libx264 -crf:v 22 video.mkv
```

And after the whole thing was dumped and dump-gnash had exited:
```
ffmpeg -i video.mkv -i audio.wav -vcodec copy -acodec libfdk_aac -ab 128k out.mkv
```
…or something of that nature.

So basically, I think he's asking for a frame dumper?

**Comment by dbluelle** [comment]
Currently lightspark doesn't have anything like that, but if anyone will implement it, I'll be happy to include that code...
